src/mock-event-bus/index.ts

```diff
@@ -29,12 +35,15 @@ export class MockEventBus implements EventBus {
      * @memberof MockEventBus
      */
     public async publish<T extends object>(event: Event<T>): Promise<boolean> {
-        return this.queues
-            .flatMap(queues => Option.of(queues.get(`${event.eventType}`)))
-            .map(fn => {
-                return fn(event);
-            })
-            .getOrElse(false);
+        return (
+            this.queues
+                // Why are we flatMapping this.queues?
+                .flatMap(queues => Option.of(queues.get(`${event.eventType}`)))
```

Comment:
every time I see `flatMap` I get suspicious that monads are involved https://www.cocoawithlove.com/blog/an-aside-about-flatmap-and-monads.html